### PR TITLE
 Configure project so react-native-web npm lib can be used in reagent forms

### LINKED ISSUE #1: Configure project so react-native-web npm lib can be used in reagent forms
https://github.com/necolas/react-native-web

You can find how we use this lib in status-electron app 
https://github.com/status-im/status-electron/blob/master/src_front/status_desktop_front/react_native_web.cljs
https://github.com/status-im/status-electron/blob/master/src_front/status_desktop_front/ui/screens/views.cljs#L3

It should be possible to render same views in web project

### PR DESCRIPTION
**Pull request by foopang**
Fix #1 

@flexsurfer Since there is no CLJS binding yet for react-native-web, I used this as reference http://blob.tomerweller.com/reagent-import-react-components-from-npm

Please advise if it is good enough for the use case.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -31,13 +31,11 @@
...
                     :source-map-timestamp true
-                    :install-deps         true
-                    :npm-deps             {:react            "15.6.1"
-                                           :react-dom        "15.6.1"
-                                           :react-native-web "0.0.100"}
                     :preloads             [devtools.preload
                                            re-frisk.preload]
-                    :external-config      {:devtools/config {:features-to-install :all}}}}
+                    :external-config      {:devtools/config {:features-to-install :all}}
+                    :foreign-libs         [{:file "public/js/bundle.js"
```
Comment (pesterhazy) [inline project.clj]: :+1:

File: src/cljs/status_fiddle/react_native_web.cljs [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
+  (cond (exists? js/ReactNative) js/ReactNative
+        (exists? js/require) (or (js/require "react-native-web")
+                                 (throw (js/Error. "require('react-native-web') failed")))
+        :else (or (aget js/window "deps" "react-native-web")
+                  (throw (js/Error. "react-native-web is missing")))))
 
-    (defn get-react-property [name]
-      (aget react-native-web name))
+(defn get-react-property [name]
+  (aget react-native-web name))
```
Comment (pesterhazy) [inline src/cljs/status_fiddle/react_native_web.cljs]: goog.object/get is preferred these days

**Comment by flexsurfer** [comment]
thanks @foopang! testing, it looks like it works :)

File: src/js/main.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+window.deps = {
+    'react' : require('react'),
+    'react-dom' : require('react-dom'),
+    'react-native-web' : require('react-native-web'),
+};
+
+window.React = window.deps['react'];
+window.ReactDOM = window.deps['react-dom'];
```
Comment (flexsurfer) [inline src/js/main.js]: why don't we have `window.ReactNativeWeb`?

**Comment by foopang** [comment]
Updated :)

File: src/cljs/status_fiddle/react_native_web.cljs [hunk 4]
```diff
@@ -1,26 +1,26 @@
...
-            [react-native-web :refer [View]]))
+  (:require [reagent.core :as reagent]))
 
-#_(defonce react-native-web
-    (cond (exists? js/ReactNative) js/ReactNative
-          (exists? js/require) (or (js/require "react-native-web")
-                                   (throw (js/Error. "require('react-native-web') failed")))
-          :else (throw (js/Error. "react-native-web is missing")))
+(defonce react-native-web
+  (cond (exists? js/ReactNative) js/ReactNative
```
Comment (flexsurfer) [inline src/cljs/status_fiddle/react_native_web.cljs]: so you can use it here just `(defonce react-native-web js/ReactNativeWeb)` if you added `window.ReactNativeWeb = window.deps['react-native-web'];`

File: src/cljs/status_fiddle/react_native_web.cljs [hunk 5]
```diff
@@ -1,26 +1,26 @@
...
+        (exists? js/require) (or (js/require "react-native-web")
+                                 (throw (js/Error. "require('react-native-web') failed")))
+        :else (or (aget js/window "deps" "react-native-web")
+                  (throw (js/Error. "react-native-web is missing")))))
 
-    (defn get-react-property [name]
-      (aget react-native-web name))
+(defn get-react-property [name]
+  (goog.object/get react-native-web name))
```
Comment (flexsurfer) [inline src/cljs/status_fiddle/react_native_web.cljs]: or better here `(goog.object/get js/ReactNativeWeb name))`

File: src/cljs/status_fiddle/react_native_web.cljs (hunk 4 above)
Comment (flexsurfer) [reply src/cljs/status_fiddle/react_native_web.cljs]: so we don't need this `defonce react-native-web`

**Comment by foopang** [comment]
@flexsurfer @pesterhazy thanks for reviewing! I've made the necessary changes accordingly.

**Comment by foopang** [comment]
Sorry, wasn't aware that there is `min` CLJS build needs for the update as well. I fixed in this commit ca3e60f3801d23b246aa8145199ca722284b9072 with a slightly different approach - to keep one build ID only and use profiles to control CLJS build settings, because I wanted to avoid having to duplicate some common parts of CLJS build settings for each build. The common parts of CLJS build settings have been extracted as one `app` build, all profile section CLJS build settings are built on top of it. Not sure if it is a good practice, it works like this now `lein with-profile <profile-id> ...`

**Comment by flexsurfer** [comment]
thanks @foopang , could you please add aliases for dev and prod? `lein figwheel-repl` and `lein build-prod` and add this info into README, thanks

**Review by flexsurfer — APPROVED** [review]

**Comment by foopang** [comment]
@flexsurfer Updated. Thanks! You merged so quickly :)

**Comment by ghost** [comment]
I think Readme.md is missing Webpack documentation. It just does not run with 'lein figwheel-repl' after cloning.
  
sudo npm install -g webpack
webpack -p

don't worry. I will add it to my commit.

**Comment by foopang** [comment]
@poiga thanks for noticing! I submitted a PR for the missing setup https://github.com/status-im/status-fiddle/pull/4

**Pull request merged**